gramps/gen/filters/rules/place/_withinarea.py

```diff
@@ -107,25 +107,22 @@ def prepare(self, db, user):
             self.radius = self.radius / 2
 
     def apply(self, dummy_db, place):
-        if self.handle is None:
-            return False
-        if self.latitude is None:
-            return False
-        if self.longitude is None:
+        if not (place and self.handle and self.latitude and self.longitude):
             return False
         if place:
-            lat = place.get_latitude()
-            lon = place.get_longitude()
+            lat: str = place.get_latitude()
+            lon: str = place.get_longitude()
```

Comment:
I'm not sure that you need these hints if `get_latitude` has the right return type hinted, which seems to have been done below here? What you probably need is to annotate the method arguments so it knows `place` is a `Place`.

Reply:
Explicit hint removed in favour of inferred type. Thanks